src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs

```diff
@@ -216,19 +212,23 @@ public AzureSqlDatabaseModel(string resourceGroup, string serverName, Management
             CreateMode = database.CreateMode;
             EarliestRestoreDate = database.EarliestRestoreDate;
 
-            CurrentServiceObjectiveId = database.CurrentServiceObjectiveId.Value;
+            CurrentServiceObjectiveName = database.CurrentServiceObjectiveName;
 
             DatabaseId = database.DatabaseId.Value;
 
             Enum.TryParse<DatabaseEdition>(database.Edition, true, out edition);
             Edition = edition;
 
-            RequestedServiceObjectiveId = database.RequestedServiceObjectiveId.Value;
+            RequestedServiceObjectiveName = database.RequestedServiceObjectiveName;
 
             Enum.TryParse<DatabaseReadScale>(database.ReadScale.ToString(), true, out readScale);
             ReadScale = readScale;
 
             ZoneRedundant = database.ZoneRedundant;
+
+            Sku = database.Sku;
+
+            Capacity = database.Sku.Capacity;
```

Comment:
@payiAzure do we need to do a null-check on `database.Sku` before accessing the `Capacity` property?